src/api.py

```diff
@@ -0,0 +1,28 @@
+import requests 
+import hashlib
+
+
+class response:
+    def __init__(self, filename, type_of_data, size, url, checksum):
+        self.filename = filename
+        self.type_of_data = type_of_data
+        self.size=size
+        self.url=url
+        self.checksum=checksum
+
+
+def fetch(url):
+    try:
+        con = requests.head(url)
+        r = requests.get(url)
+        for i in range(len(url.split('/'))-1):
+            name = url.split('/')[-(i+1)]
+            if not(name =='' ):
+                break
+        filename = '../download/'+name
+        with open(filename,'wb') as f:
+            f.write(r.content)
+        resp = response(filename, con.headers['content-type'], con.headers['content-length'],url,0)
```

Comment:
Please check you pep8 formatting and prefer using more explicit and readable keyword arguments than positional args.

Reply:
https://github.com/nexB/fetchcode/pull/4#discussion_r333589874 should I generate a name on my own or take this is a parameter from user ?